cli/docker_validator.py

```diff
@@ -48,19 +53,33 @@ def run_command_with_output(command_description, command, return_process_output=
 
 class DockerValidator(Reporter):
 
-    def __init__(self, mapping_file, output_dir, metadata_json=None,
-                 metadata_xlsx=None, container_name=container_image, docker_path='docker'):
+    def __init__(self, mapping_file, output_dir, metadata_json=None, metadata_xlsx=None,
+                 container_name=container_image, docker_path='docker', submission_config=None):
         self.docker_path = docker_path
         self.mapping_file = mapping_file
         self.metadata_json = metadata_json
         self.metadata_xlsx = metadata_xlsx
         self.container_name = container_name
         self.spreadsheet2json_conf = os.path.join(ETC_DIR, "spreadsheet2json_conf.yaml")
+        if submission_config:
+            self.sub_config = submission_config
+        else:
+            config_file = os.path.join(output_dir, SUB_CLI_CONFIG_FILE)
+            self.sub_config = WritableConfig(config_file, version=__version__)
         super().__init__(self._find_vcf_file(), output_dir)
 
     def _validate(self):
         self.run_docker_validator()
 
+    def update_config_with_validation_result(self):
+        self.sub_config.set(VALIDATION_RESULTS, value=self.results)
+        self.sub_config.set(READY_FOR_SUBMISSION_TO_EVA, value=self.verify_ready_for_submission_to_eva())
+        self.sub_config.write()
```

Comment:
I think I would model the writing of the config to the way it works in [ELOAD](https://github.com/EBIvariation/eva-submission/blob/ee152cef4cc885f74ac7f7f68971c8e9a30e6420/eva_submission/eload_submission.py#L54)

Reply:
updated